src/Ocelot/Configuration/Repository/ConsulFileConfigurationPollerOption.cs

```diff
@@ -18,7 +18,7 @@ private int GetDelay()
         {
             var delay = 1000;
 
-            var fileConfig = Task.Run(async () => await _fileConfigurationRepository.Get()).Result;
+            var fileConfig = _fileConfigurationRepository.Get().GetAwaiter().GetResult();
```

Comment:
We have this sync version in the interface:
https://github.com/ThreeMammals/Ocelot/blob/0379c7778f05b77aeea8c5e9f4512a1ea2e30ce8/src/Ocelot/Configuration/Repository/IFileConfigurationPollerOptions.cs#L5
Why not to add async version to the interface?
After that we can make private and public methods as true async &rarr; `Task<int> DelayAsync { get; }`
```suggestion
            var fileConfig = await _fileConfigurationRepository.Get();
```
Optional goal is using `Task.Delay`, if applicable.

Reply:
The expression has been revised to execute synchronously within the synchronous `int GetDelay()` method.
The proposed suggestion for refactoring is beyond the scope of this pull request. We will refactor the interface in future.